coastguard/clustersetnetworkpolicy.md

```diff
@@ -0,0 +1,272 @@
+# ClusterSetNetworkPolicy: An Approach to Multi-Cluster Network Policy
+
+Epic: [Multi cluster network policies](https://github.com/submariner-io/enhancements/issues/176)
+
+## Summary
+
+Kubernetes provides a mechanism for policing network access to pods - the [NetworkPolicy].
+The default mechanism allows specifying ingress and egress traffic rules for a set of peer pods (determined by a selector).
+A user could define network policies to allow traffic across the clusterset, by specifying peers using "ip blocks",
+but it would be impossible to use more abstract selectors to identify peers, as each cluster can only account for its own resources.
+
+We propose a new `ClusterSetNetworkPolicy` CRD for specifying multi cluster network policies.
+The CRD will have the same structure and fields as the default Kubernetes `NetworkPolicy`,
+with additional fields that are relevant to the clusterset.
+Each such policy will be translated by the local CoastGuard controller to one (or more) `NetworkPolicy` capable of allowing remote traffic.
+These local policies will be updated by the controller in response to any change in the original `ClusterSetNetworkPolicy` or any remote
+pod or namespace that are relevant to the policy.
+
+## Proposal
+
+### Problem Definition
+
+Kubernetes provides a mechanism for policing network access to pods - the [NetworkPolicy].
+The default mechanism allows specifying ingress and egress traffic rules for a set of peer pods (determined by a selector).
+The source and destination peer can be either simple IP CIDRs, or a more user friendly abstraction which allows specifying pods or
```

Comment:
L23-24 are a duplicate of L7-8.

```suggestion
The source and destination peers associated with a [NetworkPolicy] can be either simple IP CIDRs, or a more user friendly abstraction which allows specifying pods or
```

Reply:
L7-L10 is a summary of this entire section, some sentences are taken as is and some are shortened.
If anything, we can perhaps shorten the summary some more, but not this section.